compiler/pkg/openapi_generator/generator_test.go

```diff
@@ -460,4 +466,192 @@ var _ = Describe("Generator", func() {
 		Expect(gen.UpdateYAMLs(tmpDir)).To(Succeed())
 		compareTmpFileWithExpectedFile(tmpFile, "test_data/08_kubernetes_flags.yaml")
 	})
+
+	Context("checks backward compatibility", func() {
+		It("should check for incompatible changes when forcing flag enable/disable between new and existing CRDs", func() {
+			rawDefs := map[string]common.OpenAPIDefinition{
+				getSchemaName("foo"): {
+					Schema: spec.Schema{
+						SchemaProps: spec.SchemaProps{
+							Type: []string{"object"},
+							Properties: map[string]spec.Schema{
+								"spec": {
+									SchemaProps: spec.SchemaProps{
+										Type: []string{"object"},
+										Properties: map[string]spec.Schema{
+											"changePassword": {
+												SchemaProps: spec.SchemaProps{
+													Type:   []string{"string"},
+													Format: "string",
+												},
+											},
+											"name": {
+												SchemaProps: spec.SchemaProps{
+													Type:   []string{"integer"},
+													Format: "int32",
+												},
+											},
+										},
+										Required: []string{"changePassword"},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			gen, err := generator.NewGenerator(rawDefs)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(gen.ResolveRefs()).To(Succeed())
+
+			// create a new CRD with following changes
+			// removing a required field `password`.
+			// adding a new field `changePassword`
+			// modifying the `name` type from `string` to `int32`
+			tmpFile := fmt.Sprintf("%s/%s.yaml", tmpDir, "foos")
+			err = os.WriteFile(tmpFile, []byte(newFooCRD), 0665)
+			Expect(err).NotTo(HaveOccurred())
+
+			// should fail when change in annotation
+			fooContent, err := os.ReadFile(tmpFile)
+			Expect(err).To(BeNil())
```

Comment:
Can we break this single testcase into multiple cases ?
The error reporting will not be useful otherwise. Ideally, if there is a failure, the description of the testcase should tell us what has failed.
If we club all cases in a single testcase, we loose that ability.

Reply:
granularized the test case